client/packet.go

```diff
@@ -0,0 +1,109 @@
+package client
+
+import (
+	"math/rand"
+	"net"
+
+	"github.com/google/netstack/tcpip"
+	"github.com/google/netstack/tcpip/header"
+	"github.com/u-root/dhcp6"
+)
+
+func newSolicitOptions(mac net.HardwareAddr) (dhcp6.Options, error) {
+	options := make(dhcp6.Options)
+
+	// TODO: This should be generated.
+	id := [4]byte{'r', 'o', 'o', 't'}
+	// IANA = requesting a non-temporary address.
+	if err := options.Add(dhcp6.OptionIANA, dhcp6.NewIANA(id, 0, 0, nil)); err != nil {
+		return nil, err
+	}
+	// Request an immediate Reply with an IP instead of an Advertise packet.
+	if err := options.Add(dhcp6.OptionRapidCommit, nil); err != nil {
+		return nil, err
+	}
+	if err := options.Add(dhcp6.OptionElapsedTime, dhcp6.ElapsedTime(0)); err != nil {
+		return nil, err
+	}
+
+	oro := dhcp6.OptionRequestOption{
+		dhcp6.OptionDNSServers,
+		dhcp6.OptionDomainList,
+		dhcp6.OptionBootFileURL,
+		dhcp6.OptionBootFileParam,
+	}
+	if err := options.Add(dhcp6.OptionORO, oro); err != nil {
+		return nil, err
+	}
+
+	if err := options.Add(dhcp6.OptionClientID, dhcp6.NewDUIDLL(6, mac)); err != nil {
+		return nil, err
+	}
+	return options, nil
+}
+
+func newSolicitPacket(mac net.HardwareAddr) (*dhcp6.Packet, error) {
+	options, err := newSolicitOptions(mac)
+	if err != nil {
+		return nil, err
+	}
+
+	return &dhcp6.Packet{
+		MessageType: dhcp6.MessageTypeSolicit,
+		// TODO: This should be random?
+		TransactionID: [3]byte{0x00, 0x01, 0x02},
```

Comment:
Definitely feels like this should be randomized, otherwise all solicits end up with the same transaction ID.  Been a while since I've looked at DHCPv6, but this feels like it could be problematic.